src/lib/chips/chip.spec.ts

```diff
@@ -110,7 +120,132 @@ describe('Chips', () => {
+          const SPACE_EVENT: KeyboardEvent = new FakeEvent(SPACE) as KeyboardEvent;
+          const CHIP_EVENT: MdChipEvent = {chip: chipInstance};
+
+          spyOn(testComponent, 'chipSelect');
+          spyOn(testComponent, 'chipDeselect');
+
+          // Use the spacebar to select the chip
+          chipInstance._handleKeydown(SPACE_EVENT);
+          fixture.detectChanges();
+
+          expect(chipInstance.selected).toBeTruthy();
+          expect(testComponent.chipSelect).toHaveBeenCalledTimes(1);
+          expect(testComponent.chipSelect).toHaveBeenCalledWith(CHIP_EVENT);
+
+          // Use the spacebar to deselect the chip
+          chipInstance._handleKeydown(SPACE_EVENT);
+          fixture.detectChanges();
+
+          expect(chipInstance.selected).toBeFalsy();
+          expect(testComponent.chipDeselect).toHaveBeenCalledTimes(1);
+          expect(testComponent.chipDeselect).toHaveBeenCalledWith(CHIP_EVENT);
+        });
+      });
+
+      describe('when selectable is false', () => {
+        beforeEach(() => {
+          testComponent.selectable = false;
+          fixture.detectChanges();
+        });
+
+        it('SPACE ignores selection', () => {
+          const SPACE_EVENT: KeyboardEvent = new FakeEvent(SPACE) as KeyboardEvent;
+
+          spyOn(testComponent, 'chipSelect');
+
+          // Use the spacebar to attempt to select the chip
+          chipInstance._handleKeydown(SPACE_EVENT);
+          fixture.detectChanges();
+
+          expect(chipInstance.selected).toBeFalsy();
+          expect(testComponent.chipSelect).not.toHaveBeenCalled();
+        });
+      });
+
+      describe('when removable is true', () => {
+        beforeEach(() => {
+          testComponent.removable = true;
+          fixture.detectChanges();
+        });
+
+        it('DELETE emits the (remove) event', () => {
+          const DELETE_EVENT: KeyboardEvent = new FakeEvent(DELETE) as KeyboardEvent;
+
+          spyOn(testComponent, 'chipRemove');
+
+          // Use the delete to remove the chip
+          chipInstance._handleKeydown(DELETE_EVENT);
+          fixture.detectChanges();
+
+          expect(testComponent.chipRemove).toHaveBeenCalled();
+        });
+
+        it('BACKSPACE emits the (remove) event', () => {
+          const BACKSPACE_EVENT: KeyboardEvent = new FakeEvent(BACKSPACE) as KeyboardEvent;
+
+          spyOn(testComponent, 'chipRemove');
+
+          // Use the delete to remove the chip
+          chipInstance._handleKeydown(BACKSPACE_EVENT);
+          fixture.detectChanges();
+
+          expect(testComponent.chipRemove).toHaveBeenCalled();
+        });
+      });
+
+      describe('when removable is false', () => {
+        beforeEach(() => {
+          testComponent.removable = false;
+          fixture.detectChanges();
+        });
+
+        it('DELETE does not emit the (remove) event', () => {
+          const DELETE_EVENT: KeyboardEvent = new FakeEvent(DELETE) as KeyboardEvent;
```

Comment:
double cast shouldn't be necessary (also above).

TypeScript automatically interferes the types when doing `const event = new FakeEvent(DELETE) as KeyboardEvent;`